### PR TITLE
feat(types): implement RFC 4120 + RFC 6113 ASN.1 types

### LINKED ISSUE #8: feat: ASN.1 type definitions for RFC 4120 + RFC 6113
## Summary

Implement all Kerberos V5 ASN.1 types using `rasn` derive macros (Milestone 1.1 from ROADMAP).

## Scope

- 50+ ASN.1 types from RFC 4120 Appendix A + FAST (RFC 6113)
- 7 enum types with `TryFrom<i32>` (NameType, EncType, CksumType, etc.)
- 3 bitflags types (KdcOptions, TicketFlags, ApOptions)
- Type aliases (HostAddresses, AuthorizationData, MethodData, EtypeInfo, EtypeInfo2)
- `PrincipalName`: Display + FromStr + PartialEq/Eq/Hash
- `EncryptionKey`: Zeroize on drop
- Error types with 40+ KDC error code constants
- CI: cargo check, test, clippy, fmt

## Testing

- 62 round-trip DER encode/decode tests
- APPLICATION tag verification
- Flag/enum conversion tests
- BER cross-decode test

## Dependencies updated

- `rasn` 0.22 → 0.28
- Removed abandoned `rasn-kerberos`
- Added: `zeroize`, `bitflags`, `sha1`, `pbkdf2`, `subtle`
- `tokio` moved to optional (ADR-009: runtime-agnostic)
- `rand` 0.8 → 0.9

### PR DESCRIPTION
**Pull request by polaz**
## Summary

- Implement 50+ Kerberos ASN.1 types with `rasn` 0.28 derive macros
- 7 enum types (`NameType`, `EncType`, `CksumType`, etc.) with `TryFrom<i32>`
- `KdcOptions`, `TicketFlags`, `ApOptions` bitflags with `to/from_bytes`
- `PrincipalName`: `Display`, `FromStr` (with input validation), `PartialEq`/`Eq`/`Hash`
- `EncryptionKey`: key material in `Zeroizing<Vec<u8>>` with custom `Encode`/`Decode`; `Debug` redacts key bytes
- `KdcReqBody.till`: mandatory `KerberosTime` per RFC 4120
- Expanded `Krb5Error` with KDC errors, clock skew, referrals, ASN.1 errors
- Type aliases: `HostAddresses`, `AuthorizationData`, `MethodData`, `EtypeInfo`, `EtypeInfo2`
- Update deps: rasn 0.28, rand 0.9, tokio optional; remove abandoned `rasn-kerberos`
- Add `zeroize`, `bitflags`, `sha1`, `pbkdf2`, `subtle` dependencies
- CI: bump `actions/checkout` to v6

## Security

- `EncryptionKey` stores key bytes in `Zeroizing<Vec<u8>>` — guaranteed zeroed on drop
- Custom `rasn::Encode`/`Decode` impls convert to/from `OctetString` only during serialization
- `Debug` impl prints `<N bytes redacted>` instead of key material
- `FromStr` rejects malformed principals: multiple `@`, empty components, trailing/leading slashes
- `try_general_string()` propagates errors instead of silently falling back to empty string

## Testing

- **66 unit tests** — round-trip DER encode/decode for every type
- APPLICATION tag byte verification (0x61 for Ticket, 0x6a for AS-REQ, etc.)
- BER cross-decode compatibility
- Flag conversion, enum edge cases, `FromStr`/`Display` roundtrips
- `EncryptionKey` zeroize verification
- Malformed principal rejection tests

Closes #8

## Summary by CodeRabbit

* **New Features**
  * Vastly expanded Kerberos ASN.1 types, enums and primitives (AP, KDC, FAST, tickets, preauth, flags, etc.).

* **Bug Fixes**
  * Reworked error model with richer, structured error variants and improved messaging.

* **Refactor**
  * Public surface reduced: client/crypto/gssapi interfaces removed; focus on types and errors.

* **Chores**
  * Dependency updates, CI checkout action version bumped, updated ignore patterns.

* **Tests**
  * Large round‑trip encoding/decoding test suite added.

### CODE REVIEW

File: src/types/basic.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+/// Convert a string slice to GeneralString.
+///
+/// Kerberos principal names are ASCII, so this conversion is infallible
+/// in practice. Falls back to empty string on conversion failure.
+fn general_string_from(s: &str) -> KerberosString {
+    GeneralString::from_bytes(s.as_bytes())
+        .unwrap_or_else(|_| GeneralString::from_bytes(b"").expect("empty bytes are valid"))
+}
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/types/basic.rs]: Fixed — `general_string_from` replaced with `try_general_string` that returns `Result`. FromStr propagates the error; constructors use `expect()` with descriptive messages.

**Comment by polaz** [comment]
### Re: latest nitpicks (round 3)

**N8 (move zeroize import to top-level):** Fixed in 74ac899.

**N9 (ReplyValidation: `&'static str` → `Cow<'static, str>`):** Keeping `&'static str` for now — all validation failure messages are compile-time constants (`"nonce mismatch"`, `"server principal mismatch"`, etc. per RFC 4120 §3.1.5). Will change to `Cow` if dynamic messages are needed when the protocol layer is implemented.

**Copilot thread (default features):** Fixed — removed `client` and `gssapi` from default features since modules aren't implemented yet.
